design/pipeline-artifacts.md

```diff
@@ -159,6 +245,26 @@ You can give an artifact a name, and you can upload multiple named artifacts. Al
     name: WebApp
   - download:
     name: MobileApp
-  - script: TODO-some-cool-deploy-script-here $(Pipeline.ResourcesDirectory)/WebApp/bin/
-  - script: TODO-xamarin-magic $(Pipeline.ResourcesDirectory)/MobileApp/
-```
\ No newline at end of file
+  - script: ./my-deploy-script.sh $(Pipeline.Workspace)/WebApp/
+  - script: ./my-xamarin-script.sh $(Pipeline.Workspace)/MobileApp/
+```
+
+### Multiple artifact downloads with explicit path
+
+If a download step covers multiple artifacts and no explicit path is given, they're each pulled into a separate directory. With an explicit path, that explicit path becomes a containing folder for the artifacts.
```

Comment:
Would it be simpler to explain and more predictable, to say that when an artifact name and path is specified, the artifact contents are downloaded to that folder. Otherwise artifacts are downloaded into a folder named using the artifact name.

Customers deserve a simple rule, for explicit control. And consistent behavior otherwise, that meets the mainline needs.

Reply:
Simplifying the behavior.